flytestdlib/cache/auto_refresh.go

```diff
@@ -173,6 +175,25 @@ func (w *autoRefresh) Start(ctx context.Context) error {
 	return nil
 }
 
+// Update updates the item only if it exists in the cache, return true if we updated the item.
+func (w *autoRefresh) Update(id ItemID, item Item) (ok bool) {
+	w.lock.Lock()
+	defer w.lock.Unlock()
+	ok = w.lruMap.Contains(id)
+	if ok {
+		w.lruMap.Add(id, item)
```

Comment:
Why only update if it's already there, what if it has been evicted since?

Reply:
If we don't check if it exist in the cache, we might have this scenario.

`worker1` removes item from the caches -> `worker2` adds the item to cache again. (workerqueue can have duplicate item)

If we don't use lock, we might have this scenario.

1. w.toDelete.Remove (worker 1)
2. w.lruMap.Contains(id) (woker 2)
3. w.lruMap.Remove(key) (woker 1)
4. w.lruMap.Add(id, item) (woker 2) # we add item back to the cache again.